autopush/router/gcm.py

```diff
@@ -98,12 +99,14 @@ def _route(self, notification, router_data):
             data['con'] = con
             data['enc'] = enc
 
+        # registration_ids are the GCM instance tokens (specified during
+        # registration.
         payload = gcmclient.JSONMessage(
-            registration_ids=[router_data["token"]],
+            registration_ids=[router_data.get("token")],
             collapse_key=self.collapseKey,
             time_to_live=self.ttl,
-            dry_run=self.dryRun,
-            data=udata,
+            dry_run=self.dryRun or "dryrun" in router_data,
```

Comment:
Remembering the order of operation precedence isn't my strong-point so I usually () clauses like this just to be sure. Nick actually found an error in older GCM code due to operator precedence working in an unintentional way.